### PR TITLE
NO_TICKET ensure transfers are included in block limited only by bloc…

### PR DESCRIPTION
**Pull request by dwerner**
This PR is intended to ensure that transfers are included in block limited only by `block_max_transfer_limit`. @EdHastingsCasperLabs asked for this change in our testing meeting today.

### CODE REVIEW

**Comment by goral09** [comment]
What's the size native transfer deploy?

File: node/src/components/block_proposer/tests.rs [hunk 1]
```diff
@@ -585,6 +567,8 @@ fn test_proposer_with(
     let block = proposer.propose_proto_block(config, test_time, past_deploys, true);
     let all_deploys = BlockLike::deploys(&block);
     proposer.finalized_deploys(all_deploys.iter().map(|hash| **hash));
+    println!("proposed deploys {}", block.wasm_deploys().len());
+    println!("proposed transfers {}", block.transfers().len());
```
Comment (goral09) [inline node/src/components/block_proposer/tests.rs:571]: Leftover?

File: node/src/components/block_proposer/tests.rs [hunk 2]
```diff
@@ -483,24 +469,24 @@ fn should_not_propose_deploy_if_block_size_limit_within_threshold() {
...
     });
 }
 
 #[test]
 fn should_not_propose_deploy_if_block_size_limit_passed() {
     test_proposer_with(TestArgs {
-        deploy_count: 0,
-        transfer_count: 1,
+        deploy_count: 3,
+        transfer_count: 2, // transfers should -not- count towards the block size limit
```
Comment (goral09) [inline node/src/components/block_proposer/tests.rs:482]: Can you add a test that sets a block size limit lower than `max_transfer_count` occupies? i.e. show that transfers can exceed block size limit.

File: node/src/components/block_proposer/tests.rs (hunk 1 above)
Comment (dwerner) [reply node/src/components/block_proposer/tests.rs:571]: I'd like to keep these, as they are nice to have as output in the tests if they fail.

**Comment by dwerner** [comment]
> What's the size of native transfer deploy?

There is a constant in `block_proposer` I added:
```
/// Experimentally, deploys are in the range of 270-280 bytes, we use this to determine if we are
/// within a threshold to break iteration of `pending` early.
const DEPLOY_APPROX_MIN_SIZE: usize = 300;
```

File: node/src/components/block_proposer/event.rs [hunk 3]
```diff
@@ -72,7 +72,7 @@ impl DeployType {
     }
 
     /// Asks if the variant is Wasm.
-    pub fn is_wasm(&self) -> bool {
+    pub fn is_other(&self) -> bool {
```
Comment (goral09) [inline node/src/components/block_proposer/event.rs:75]: `is_wasm` was a better name IMO. I understand that the Wasm variants is called `Other` right now but in our domain it's either Wasm or native transfer and I would stick to these names for now. When we get to add other variants w can change names, add other methods.

Comment (dwerner) [reply node/src/components/block_proposer/event.rs:75]: Fair enough, I'll change it back. :+1:

File: node/src/components/block_proposer/tests.rs (hunk 1 above)
Comment (EdHastingsCasperLabs) [reply node/src/components/block_proposer/tests.rs:586]: Then change the logic to only print them when it fails, please.

**Review by EdHastingsCasperLabs — APPROVED** [review]
nitpick should not block merge as we need this urgently

**Pull request merged**